test/suites/rt-upgrade-chopsticks-orchestrator/test-upgrade-chain.ts

```diff
@@ -11,17 +19,77 @@ describeSuite({
     testCases: ({ it, context, log }) => {
         let api: ApiPromise;
         let specName: string;
+        let alice: KeyringPair;
+        let runtimeName: String;
+        let xcmQueryToAnalyze: Number;
 
         beforeAll(async () => {
             api = context.pjsApi;
             specName = api.consts.system.version.specName.toString();
             const specVersion = getSpecVersion(api);
             log(`Currently connected to chain: ${specName} : ${specVersion}`);
+            runtimeName = api.runtimeVersion.specName.toString();
+
+            const keyring = new Keyring({ type: "sr25519" });
+            context.keyring.alice;
+            alice = keyring.addFromUri("//Alice", { name: "Alice default" });
+
+            if (runtimeName != "flashbox") {
+                // Inject a query so that there are migrations to execute in queries in case of new xcm version
+                // Right now we need to hardcode the xcm version
+                // this should not be necessary after we bring https://github.com/paritytech/polkadot-sdk/pull/8173
+                // since we would be able to fetch it on-chain
+                const queryLocation = runtimeName.includes("light")
+                    ? {
+                          parents: 0,
+                          interior: { X1: [{ Parachain: 1 }] },
+                      }
+                    : {
+                          parents: 1,
+                          interior: "Here",
+                      };
+
+                // fetch on-chain later
+                // TODO:Once we update the next runtime, remove this and take it form onchain
+                const previousXcmVersion = 5;
+                const latestVersion = "V" + previousXcmVersion.toString();
+
+                const versionedLocation = new Object();
```

Comment:
It's a bit outdated way to create objects in JS.
You can change it to something like this:
```suggestion
const versionedLocation = {
  [latestVersion]: queryLocation
};
```